src/Symfony/Component/RateLimiter/Policy/FixedWindowLimiter.php

```diff
@@ -69,15 +69,17 @@ public function reserve(int $tokens = 1, float $maxTime = null): Reservation
                 $reservation = new Reservation($now, new RateLimit($window->getAvailableTokens($now), \DateTimeImmutable::createFromFormat('U', floor($now)), true, $this->limit));
             } else {
                 $waitDuration = $window->calculateTimeForTokens($tokens);
+                $timerDate = \DateTimeImmutable::createFromFormat('U.u', $window->getTimer());
+                $retryAfterDate = $timerDate->modify('+'.ceil($window->getExpirationTime()).' seconds');
```

Comment:
@BrokenSourceCode, is it the same as the following (that I find a bit more readable)?
```
$retryAfterDate = \DateTimeImmutable::createFromFormat('U.u', $window->getTimer() + $window->getExpirationTime());
```
Also, `$window->getExpirationTime()` returns `int`, so `ceil()` only casts the returned `int` to `float`.

Reply:
Sure, that's better